encoding/marshal_test.go

```diff
@@ -216,18 +216,58 @@ func TestEncodeDecode(t *testing.T) {
 	}
 }
 
+func TestEncodeAll(t *testing.T) {
+	var expected []byte
+	for i := range testStructs {
+		expected = append(expected, Marshal(testStructs[i])...)
+	}
```

Comment:
hmm. Since `EncodeAll` is really just a sequence of encodes, maybe I should use `Encode` instead of marshal.
A hardcoded test wouldn't hurt as well.